src/main/java/org/truffleruby/language/loader/SourceLoader.java

```diff
@@ -140,11 +140,15 @@ public Source load(String canonicalPath) throws IOException {
             Log.LOGGER.info("loading " + canonicalPath);
         }
 
+        return loadNoLogging(context, canonicalPath, isInternal(canonicalPath));
+    }
+
+    public static Source loadNoLogging(RubyContext context, String canonicalPath, boolean internal) throws IOException {
```

Comment:
Is the significance of "no logging" really that the context isn't being used?

Reply:
This is used both with a null and non-null context.
The only part completely ignored in this method from the original is the logging, hence the name.
It's a static version of load() handling whether context is null or not.